analysis/src/main.js

```diff
@@ -38,17 +37,35 @@ function processTasks() {
 
   var locky = new Date().toString() + ".lock";
 
-  app.post('/process/next', jsonBodyParser, (req, res) => {
-    var message = req.body.message;
-    // force single-thread, immediately fail
-    lockfile.lock(locky, {}, err => {
+  app.get('/process/next', (req, res) => {
+    var attributes = {
+      owner: req.query.owner,
+      repo: req.query.repo,
+      version: req.query.version,
+      responseTopic: req.query.responseTopic
+    };
+    if (req.query.sha) {
+      attributes.sha = req.query.sha;
+    }
+
+    // We only accept requests from the task queue service.
+    // This check is valid because appengine strips external x-appengine headers.
+    // By definition, this is internal.
+    if (!req.get('x-appengine-queuename')) {
+      Ana.fail("main/processTasks/originator-not-appengine", JSON.stringify(attributes));
+      res.status(200).send(); // Don't retry
```

Comment:
I think `403` makes more sense here. If this ever were to happen we'd probably want to retry anyway.